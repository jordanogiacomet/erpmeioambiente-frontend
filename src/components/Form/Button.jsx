import P from 'prop-types';
import styled from 'styled-components';


const ButtonWrapper = styled.button`
    width: 100%;
    padding: 15px 0px; 
    background-color: ${({ theme }) => theme.colors.secondary};
    border: transparent;
    border-radius: 10px;
    font-size: ${({theme}) => theme.font.sizes.large};
    font-weight: 700;
    cursor: pointer;
`;


export const Button = ({ text, ...props }) => {
  return (
    <ButtonWrapper aria-label={text} {...props}>{text}</ButtonWrapper>
  );
};

Button.propTypes = {
  text: P.string.isRequired,
};

export default Button;