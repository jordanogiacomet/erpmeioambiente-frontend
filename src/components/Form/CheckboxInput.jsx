import React from 'react';
import styled from 'styled-components';
import P from 'prop-types';

const CheckboxWrapper = styled.div`
  display: flex;
  align-items: center;
`;

const CheckboxLabel = styled.label`
  margin-left: 8px;
  color: ${({ theme }) => theme.colors.secondary};
  font-size: ${({ theme }) => theme.font.sizes.large};
  font-weight: 600;
`;

const CheckboxField = styled.input`
  width: auto;
  cursor: pointer;
`;

const CheckboxInput = ({ label, id, ...props }) => (
  <CheckboxWrapper>
    <CheckboxField type="checkbox" id={id} {...props} />
    <CheckboxLabel htmlFor={id}>{label}</CheckboxLabel>
  </CheckboxWrapper>
);

CheckboxInput.propTypes = {
  label: P.string.isRequired,
  id: P.string.isRequired,
};

export default CheckboxInput;
