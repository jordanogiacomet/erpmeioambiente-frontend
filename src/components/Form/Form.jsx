import React from 'react';
import styled from 'styled-components';
import P from 'prop-types';

const FormWrapper = styled.form`
  width: 100%;
  max-width: 600px;
  margin: 0 auto;
  padding: 24px;
  background-color: ${({ theme }) => theme.colors.secondary};
  border-radius: 30px;
  box-shadow: 0 4px 12px rgba(0,0,0,0.1);
  border: 1px solid ${({ theme }) => theme.colors.primary};
  & > *:not(:last-child) {
    margin-bottom: 16px;
  };
`;

const Form = ({ children, ...props }) => (
  <FormWrapper {...props}>{children}</FormWrapper>
);

Form.propTypes = {
  children: P.node.isRequired,
};

export default Form;