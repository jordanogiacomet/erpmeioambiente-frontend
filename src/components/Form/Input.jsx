import React from 'react';
import styled from 'styled-components';
import P from 'prop-types';

const InputWrapper = styled.div`
  margin-bottom: 16px;
`;

const Label = styled.label`
  display: block;
  margin-bottom: 8px;
  color: ${({ theme }) => theme.colors.text};
`;

const InputField = styled.input`
  padding: 10px 15px;  // Ajuste o padding para um estilo mais confortÃ¡vel
  border: 2px solid ${({ theme }) => theme.colors.border};
  border-radius: 25px;  // Arredondamento para os cantos do input
  background-color: ${({ theme }) => theme.colors.secondary};
  color: ${({ theme }) => theme.colors.text};
  outline: none;
  transition: border-color 0.3s ease;

  &:focus {
    border-color: ${({ theme }) => theme.colors.focus};
  }
`;

const Input = ({ label, ...props }) => (
  <InputWrapper>
    <Label>{label}</Label>
    <InputField {...props} />
  </InputWrapper>
);

Input.propTypes = {
  label: P.string
};

export default Input;