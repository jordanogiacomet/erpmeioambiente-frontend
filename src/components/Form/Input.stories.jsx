import React from 'react';
import Input from '../Form/Input';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';

const withThemeProvider = (Story, context) => (
  <ThemeProvider theme={theme}>
    <Story {...context} />
  </ThemeProvider>
);

export default {
  title: 'Form/Input',
  component: Input,
  decorators: [withThemeProvider],
};

export const Default = (args) => <Input {...args} />;

Default.args = {
  label: 'Default Input',
  placeholder: 'Enter text...',
};

export const DefaultWithoutLabel = (args) => <Input {...args} />;

DefaultWithoutLabel.args = {
  placeholder: 'Enter text...',
};


export const Email = (args) => <Input {...args} />;

Email.args = {
  label: 'Email Input',
  type: 'email',
  placeholder: 'Enter your email',
};

export const Password = (args) => <Input {...args} />;

Password.args = {
  label: 'Password Input',
  type: 'password',
  placeholder: 'Enter your password',
};