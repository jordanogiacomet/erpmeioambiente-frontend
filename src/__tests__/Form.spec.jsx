import React from 'react';
import renderTheme from '../styles/render-theme';
import 'jest-styled-components';
import Form from '../components/Form/Form';

describe('<Form />', () => {
  it('should render correctly with inputs', () => {
    const { container } = renderTheme(
      <Form>
        <Input label="Nome" />
        <Input label="Email" type="email" />
        <button type="submit">Registrar</button>
      </Form>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '30px');
    expect(container.firstChild).toHaveStyleRule('box-shadow', '0 4px 12px rgba(0,0,0,0.1)');
  });
});