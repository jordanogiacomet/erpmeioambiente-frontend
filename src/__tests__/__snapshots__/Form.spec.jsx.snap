// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render correctly with styles 1`] = `
.c0 {
  width: 100%;
  max-width: 600px;
  margin: 0 auto;
  padding: 24px;
  background-color: #FFFFFF;
  border-radius: 30px;
  box-shadow: 0 4px 12px rgba(0,0,0,0.1);
  border: 1px solid #4169E1;
}

.c0>*:not(:last-child) {
  margin-bottom: 16px;
}

<form
  class="c0"
>
  <input
    placeholder="Test Input"
    type="text"
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
